### PR TITLE
Handle ParenthesizedType

### LINKED ISSUE #154: Optional function as an argument generates code that doesn't typecheck
Compile this:
```
const fun = (callback: ?() => void = null): void =>  {}
```
And you'll get this: 
```
const fun = (callback: () => void | null | undefined = null): void => {};
```

Generate a react app like so:
```
create-react-app --typescript myapp
```
Toss this line of code into one of the files and try to run it
```yarn start```
This happens:
```
/home/nretallack/apps/myapp/src/App.tsx
TypeScript error in /home/nretallack/apps/myapp/src/App.tsx(26,14):
Type 'null' is not assignable to type '() => void | null | undefined'.  TS2322

    24 | }
    25 | 
  > 26 | const fun = (callback: () => void | null | undefined = null): void => {};
       |              ^
    27 | 
    28 | export default App;
    29 | 
```
The error goes away if you wrap the function type in parenthesis, like this:
```
const fun = (callback: (() => void) | null | undefined = null): void => {};
```

### PR DESCRIPTION
**Pull request by cuyl**
Fixes #154

### CODE REVIEW

**Comment by cuyl** [comment]
[clabot:check]

**Comment by cuyl** [comment]
[clabot:check]

**Comment by cuyl** [comment]
@kevinbarabash

File: src/transform.js [hunk 1]
```diff
@@ -326,7 +326,14 @@ const transform = {
       }
       const typeAnnotation = t.tsTypeAnnotation(returnType);
       path.replaceWith(
-        t.tsFunctionType(typeParameters, parameters, typeAnnotation)
+        !path.parent ||
```
Comment (k4b7) [inline src/transform.js:329]: When would `path` not have a parent?

**Review by k4b7 — APPROVED** [review]
LGTM.  Thanks for fixing this.

File: src/transform.js (hunk 1 above)
Comment (cuyl) [reply src/transform.js:329]: I don't know, I guess `path.parent` can be `undefined`.

**Pull request merged**